channeldb/invoices.go

```diff
@@ -96,14 +91,40 @@ const (
 
 	// A set of tlv type definitions used to serialize invoice htlcs to the
 	// database.
+	//
+	// NOTE: A migration should be added whenever this list changes. This
+	// prevents against the database being rolled back to an older
+	// format where the surrounding logic might assume a different set of
+	// fields are known.
 	chanIDType       tlv.Type = 1
 	htlcIDType       tlv.Type = 3
 	amtType          tlv.Type = 5
 	acceptHeightType tlv.Type = 7
 	acceptTimeType   tlv.Type = 9
 	resolveTimeType  tlv.Type = 11
 	expiryHeightType tlv.Type = 13
-	stateType        tlv.Type = 15
+	htlcStateType    tlv.Type = 15
+
+	// A set of tlv type definitions used to serialize invoice bodiees.
+	//
+	// NOTE: A migration should be added whenever this list changes. This
```

Comment:
The migration would only just update the DB version to prevent a downgrade? Seems like we could also implement a run-time sanity check based on if the DB includes a type not in the set of known types, or the other way around.

Reply:
> The migration would only just update the DB version to prevent a downgrade? 

There will also be code added to read and write the new field

> Seems like we could also implement a run-time sanity check based on if the DB includes a type not in the set of known types, or the other way around.

Does that mean we'd read the entire db on every open? or only do it when the data is accessed? i think it ultimately depends on what the field is, in certain cases we could lose money if a downstream action fails that we are expecting to succeed. the migration is def the safest option because it doesn't create the illusion that things are okay